app/main.py

```diff
@@ -1,3 +1,44 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: ("Distance", int, float)) -> callable:
+        if not isinstance(other, (int, float)):
+            return Distance(self.km + other.km)
```

Comment:
The condition should check if 'other' is an instance of 'Distance' instead of 'int' or 'float'. The logic is reversed here.